feat(mme): Support for PCO with DNS option in amf

panyogesh opened the pull request:
Fix:
  Enable the encoding/decoding of the options in NAS5G
  Enabling the DNS Server configuration for AMF

Test:
  Unit testing
  Simulator testing

Logs:
  10/13 Test #10: test_amf .........................   Passed    0.06 sec

Signed-off-by: Yogesh Pandey <yogesh@wavelabs.ai>

<!--
    Tag your PR title with the components that it touches along with
    the type of change
    E.g. "fix(orc8r): Fix reindexer race condition" or "feat(agw) ..."
-->

## Summary

<!-- Enumerate changes you made and why you made them -->

## Test Plan

<!--
    How did you test your change? How do you know it works?
    Add supporting screenshots, terminal pastes, etc. as necessary
-->

## Additional Information

- [ ] This change is backwards-breaking

<!--
    If this is a backwards-breaking change, document the upgrade instructions.
    All upgrade instructions for backwards-breaking changes will be aggregated
    in the next release's changelog so this is very important to fill out.
-->

panyogesh committed:
feat(mme): Support for PCO with DNS option in amf

Fix:
  Enable the encoding/decoding of the options in NAS5G
  Enabling the DNS Server configuration for AMF

Test:
  Unit testing
  Simulator testing

Signed-off-by: Yogesh Pandey <yogesh@wavelabs.ai>

lionelgo commented on lte/gateway/c/core/oai/lib/3gpp/3gpp_24.008.h:
```diff
@@ -886,6 +886,11 @@ int decode_access_point_name_ie(
+#define IPCP_CODE_BYTES 1
+#define IPCP_IDENTIFIER_BYTES 1
+#define IPC_LENGTH_BYTES 2
+
```
Not sure IPCP_* should go into 24.008 related .h file.

panyogesh replied:
Comments addressed.

lionelgo commented on lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp:
```diff
@@ -812,13 +814,14 @@ int amf_app_handle_pdu_session_accept(
+
+  encode_msg->payload_container.len =
+      PDU_ESTAB_ACCPET_PAYLOAD_CONTAINER_LEN + pco_len;
+  len = PDU_ESTAB_ACCPET_NAS_PDU_LEN + pco_len;
```
typo accept

panyogesh replied:
Comments addressed.

lionelgo commented on lte/gateway/c/core/oai/tasks/amf/amf_app_ue_context_and_proc.h:
```diff
@@ -95,7 +95,7 @@ struct amf_procedures_t;
 #define AMBR_LEN 6
 #define PDU_ESTAB_ACCPET_PAYLOAD_CONTAINER_LEN 30
-#define PDU_ESTAB_ACCPET_NAS_PDU_LEN 39
+#define PDU_ESTAB_ACCPET_NAS_PDU_LEN 41
```
Typo accept ?

panyogesh replied:
Comments addressed.

panyogesh committed:
feat(mme): Support for PCO in PDU Session 5G

Fix:
  Extended Support for PCO in 5G

Testcase:
  Unit testcases of oai
  Test with Simulator

Signed-off-by: Yogesh Pandey <yogesh@wavelabs.ai>

panyogesh commented:
![Uploading Packet-Capture-PCO-Aug-31.png…]()

panyogesh commented:
![image](https://user-images.githubusercontent.com/69527565/131506088-82220730-bba5-460f-89eb-ce2eb934af75.png)

ulaskozat reviewed (changes requested):
conflict exists

ulaskozat reviewed (approved):
LGTM
